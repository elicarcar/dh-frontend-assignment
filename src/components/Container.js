import React from "react";
import styled from "styled-components";

const Wrapper = styled.div`
  position: fixed;
  top: 250px;
`;

const Container = props => {
  return <Wrapper>{props.children}</Wrapper>;
};

export default Container;
