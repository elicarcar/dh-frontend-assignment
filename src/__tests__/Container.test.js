import React from "react";
import Container from "../components/Container/Container.js";
import renderer from "react-test-renderer";
import "jest-styled-components";
import "@testing-library/jest-dom/extend-expect";

describe("<Container/> component", () => {
  it("renders correctly and have specified style rules", () => {
    const tree = renderer.create(<Container />).toJSON();
    expect(tree).toHaveStyleRule("position", "fixed");
    expect(tree).toHaveStyleRule("top", "250px");
    expect(tree).toMatchSnapshot();
  });
});
