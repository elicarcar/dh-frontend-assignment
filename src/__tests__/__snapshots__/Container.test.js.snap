// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Container/> component renders correctly and have specified style rules 1`] = `
.c0 {
  position: fixed;
  top: 250px;
}

<div
  className="c0"
/>
`;
